File: ice0208/two-pointers/283.py
class Solution:
    def moveZeroes(self, nums: list[int]) -> None:
        check = 0
        i = 0
        while check < len(nums):
            if nums[i] == 0:
                nums.pop(i)
                nums.append(0)
            else:
                i += 1

            check += 1

        return nums


if __name__ == "__main__":
    print(Solution().moveZeroes([0, 1, 0, 3, 12]))
    print(Solution().moveZeroes([0]))

File: moonpoison/two_pointers/283.cpp
#include<bits/stdc++.h>
using namespace std;
class Solution {
public:
    void moveZeroes(vector<int>& nums) 
    {
        int i=0;
        int flag=nums.size();
        while(flag!=0)
        {
            flag--;
            if(nums[i]==0)
            {
                nums.push_back(0);
                nums.erase(nums.begin()+i);
            }
            else ++i;
        }
    }
};
